### PR TITLE
add iOS Feature - API #37

### PR DESCRIPTION
**Pull request by KaoWeiCheng**
#### Request:
`GET https://{domainhost}/v1/boards?code={code}&limit={limit}`

`GET https://{domainhost}/v1/themes?{type}`

#### Note
The columns "created_at" and "updated_at" in the DB table "boards" correspond to "created_date" and "updated_date" mentioned in the documentation. 
My program currently prioritizes the table's columns, so if any changes are needed, please let me know. Thank you.

### CODE REVIEW

File: src/main/java/com/orange/main/Theme/service/impl/ThemeServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+public class ThemeServiceImpl implements ThemeService {
+
+    @Autowired
+    private ThemeRepository themeRepository;
+
+    @Override
+    public List<Theme> getThemeByType(String type) {
+        return themeRepository.findByType(type);
```
Comment (Vince-Ku) [inline src/main/java/com/orange/main/Theme/service/impl/ThemeServiceImpl.java]: What are the consequences of encountering an exception in this context? like DB connection failure or an object mapping failure, In such cases, What exact JSON format will be returned?

File: src/main/java/com/orange/main/boards/service/impl/BoardssServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@Service
+public class BoardssServiceImpl implements BoardsService {
+
+    @Autowired
+    private BoardsRepository boardsRepository;
+
+    @Override
+    public List<Board> getBoardsByCodeAndLimit(String code, int limit) {
+        return boardsRepository.findByCode(code, PageRequest.of(0, limit));
+    }
```
Comment (Vince-Ku) [inline src/main/java/com/orange/main/boards/service/impl/BoardssServiceImpl.java:21]: ditto

**Comment by asdfg429** [comment]
LGTM
@Vince-Ku your questions will solved after you check issue #15.
But I don't know how to link this issue to #15.
I created an ExceptionHandler.

**Comment by Vince-Ku** [comment]
> LGTM @Vince-Ku your questions will solved after you check issue #15. But I don't know how to link this issue to #15. I created an ExceptionHandler.

got it.

**Review by Vince-Ku — APPROVED** [review]
LGTM

File: src/main/java/com/orange/main/Theme/controller/ThemeController.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1")
```
Comment (asdfg429) [inline src/main/java/com/orange/main/Theme/controller/ThemeController.java]: Please remove this line. Check #15. I add some code on BaseController like these `@RequestMapping("api/v1")` .
These could make all RestControllers URL started with `api/v1` .

File: src/main/java/com/orange/main/boards/controller/BoardsController.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/v1")
```
Comment (asdfg429) [inline src/main/java/com/orange/main/boards/controller/BoardsController.java]: ditto

**Review by asdfg429 — CHANGES_REQUESTED** [review]
Check comments, please remove annotation.

File: src/main/java/com/orange/main/Theme/controller/ThemeController.java (hunk 3 above)
Comment (Vince-Ku) [reply src/main/java/com/orange/main/Theme/controller/ThemeController.java]: https://github.com/No-57/orange-frontstage-api/pull/22/commits/162cc574fcb5b4d19efcff3810d77cdbf1072a0d

File: src/main/java/com/orange/main/boards/controller/BoardsController.java (hunk 4 above)
Comment (Vince-Ku) [reply src/main/java/com/orange/main/boards/controller/BoardsController.java]: https://github.com/No-57/orange-frontstage-api/pull/22/commits/162cc574fcb5b4d19efcff3810d77cdbf1072a0d

**Comment by Vince-Ku** [comment]
@KaoWeiCheng 
I made a few adjustments, please check.

For a quick test, I've rebased this branch onto the latest Main branch, I hope this is ok with you 🥹.

**Comment by KaoWeiCheng** [comment]
No problem at all, thank you!

**Pull request merged**